packages-content-model/roosterjs-content-model-plugins/lib/edit/deleteSteps/deleteList.ts

```diff
@@ -0,0 +1,24 @@
+import { getClosestAncestorBlockGroupIndex } from 'roosterjs-content-model-core';
+import type { DeleteSelectionStep } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export const deleteList: DeleteSelectionStep = context => {
+    const { paragraph, marker, path } = context.insertPoint;
+
+    if (context.deleteResult == 'nothingToDelete' || context.deleteResult == 'notDeleted') {
+        const index = getClosestAncestorBlockGroupIndex(path, ['ListItem']);
```

Comment:
Please add the 3rd parameter "['TableCell']" so if there is table under list, do not delete list. And add test case for this scenario